metatensor-torch/tests/block.cpp

```diff
@@ -77,4 +77,15 @@ TEST_CASE("Blocks") {
             CHECK(std::get<0>(entry) == "g");
         }
     }
+
+    SECTION("different devices") {
+        REQUIRE_THROWS(
+            TensorBlockHolder(
+                torch::full({3, 2}, 11.0),
+                LabelsHolder::create({"s"}, {{0}, {2}, {1}})->to(torch::kMeta),
+                std::vector<TorchLabels>{},
+                LabelsHolder::create({"p"}, {{0}, {1}})
+            )
+        );
```

Comment:
Please use `CHECK_THROWS_WITH` here. `CHECK_` instead of `REQUIRES_` just means that the test will continue running if this specific check fails. The `_WITH` part checks the error message associated with the exception. (You might need to use `StartsWith(....)` here, see others tests for usage example)

```suggestion
        CHECK_THROWS_WITH(
            TensorBlockHolder(
                torch::full({3, 2}, 11.0),
                LabelsHolder::create({"s"}, {{0}, {2}, {1}})->to(torch::kMeta),
                std::vector<TorchLabels>{},
                LabelsHolder::create({"p"}, {{0}, {1}})
            ),
            "expected error message..."
        );
```